### ISSUE TITLE
Latest maven does not compile

### ISSUE THREAD

**Issue by slifty** [prior-art-archive/query-parser-docker#2]
Running `mvn clean install` fails when using maven 3.6.1

Here's the error it fails on 
```
ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project server: Compilation failure: Compilation failure:
[ERROR] Source option 5 is no longer supported. Use 6 or later.
[ERROR] Target option 1.5 is no longer supported. Use 1.6 or later.
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :server
```

We should document the version of maven that this project requires (and the version of Java).  If we decide that the project should support more recent versions we should create an issue to capture that task.

**Comment by slifty** [comment]
cc @shikham01 What is the version of maven and java that @reefdog and I should be using?

**Comment by slifty** [comment]
Reverting to Java 8 (from Java 11) fixed this issue; however now I'm seeing:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project server: Compilation failure
[ERROR] /Users/slifty/Dropbox/Code/Java/query-parser-docker/server/src/main/java/com/cloud/uspto/parser/query/SpanParsingRules.java:[82,26] cannot find symbol
[ERROR]   symbol:   method ppushWithContext(org.parboiled.support.Var<java.lang.String>)
[ERROR]   location: class com.cloud.uspto.parser.query.SpanParsingRules
[ERROR]
```

Will investigate.

**Comment by slifty** [comment]
Good news, looks like this may just be a typo -- `ppushWithContext` => `pushWithContext`

I'll open an issue to that effect and a separate PR / hotfix.

**Comment by slifty** [comment]
Now that we closed #4 it compiles with Java 8!  Updating the readme to reflect that requirement, but otherwise this is all set.

**Comment by slifty** [comment]
Note added (also skipped PR since it's just documentation)
